aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -57,6 +57,86 @@ TORCH_META_FUNC(mm)(const Tensor & self, const Tensor & mat2) {
   "The input tensor must be a matrix with size ", self.sizes()[0], "x", mat2.sizes()[1], ", but got a ", result.dim(),
   "-D tensor with size ", result.sizes()[0], "x", result.sizes()[1]);
 }
+
+template <typename Meta>
+void common_checks_baddbmm_bmm(Meta& meta, const Tensor& batch1, const Tensor& batch2, const Scalar& beta, const Scalar& alpha, bool is_bmm, const c10::optional<Tensor>& self_or_result=nullopt) {
+  CheckedFrom c = (is_bmm ? "bmm" : "baddbmm");
+
+  auto checkOnCPU = [](const Tensor& t, CheckedFrom c) {
+    TORCH_CHECK(
+      !t.is_cuda(),
+      "Expect tensor to have CPU backend, but got tensor with ",
+      toString(t.options().backend()),
+      " Backend (while checking arguments for ",
+      c);
+  };
+
+  if (self_or_result.has_value()) checkOnCPU(self_or_result.value(), c);
+  checkOnCPU(batch1, c);
+  checkOnCPU(batch2, c);
+
+  checkDim(c, batch1, "batch1", /* pos */1, /* dim */ 3);
+  checkDim(c, batch2, "batch2", /* pos */ 2, /* dim */ 3);
```

Comment:
These 2 checks are the same as the ones for CUDA in the `META` function, aren't they? Aren't they device-independent?

Reply:
Right, indeed they are! This is resolved now.